### DISCUSSION TITLE
Kill Notification

### DISCUSSION THREAD

**Discussion by gm90** [johnlindquist/kit#288] [Share]
Inspired by Tyler Hall, a quick shortcut key to clear a mac notification with the awkward Big Sur UX  with the tiny cross click area.

```javascript
// Shortcut: command option k
// Menu: Kill Notification
// Description: Kill pesky mac os notification inspired by 240px-invisible-pixels @ https://tyler.io/240-invisible-pixels/
// Author: Gary McCann

const robot = await npm ("robotjs");
const screenSize = robot.getScreenSize();
const xPosition = screenSize.width - 354;
const yPosition = 47;

robot.moveMouse(xPosition, yPosition);

setTimeout(() => {  
  robot.mouseClick();
}, 100)

**Comment by johnlindquist** [comment]
Lol, awesome.

I did some digging and found [an AppleScript (in JS) solution](https://gist.github.com/lancethomps/a5ac103f334b171f70ce2ff983220b4f).

I'll need to add a "JavaScript" flag to the `applescript` helper to run this code though. Or I guess we could create a `close-notifications` file like so:

```bash
#!/usr/bin/osascript -l JavaScript

// JS gist goes here

```

Then chmod +x it and run it with `exec("close-notifications")`. 🤔

**Reply by gm90** [reply]
🤔 interesting, any advantage to using the AppleScript one over that ☝️ ☝️?

Had run ins in the past with AppleScript whilst messing with BetterTouchTool so when Kit came along was able to happily avoid going back there 🤣

**Reply by johnlindquist** [reply]
Yeah, AppleScript is atrocious. I have nothing but terrible feelings towards AppleScript.

Advantages? Not much. It might be able to close all notifications in a batch. It avoids moving the mouse, but that doesn't seem like much of an issue.

Later this year I'll dig into the Rust or Swift bindings to interact with osx UI through scripts so I can remove all the AppleScript helpers. Lots of other tasks before I get there though. 🙂

**Reply by gm90** [reply]
Glad its not just me then 😊 .

Fair point about moving the mouse. Slightly modified version to put the mouse back where it started, like nothing ever happened ;) 

```javascript
// Shortcut: command option k
// Menu: Kill Notification
// Description: Kill pesky mac os notification inspired by 240px https://tyler.io/240-invisible-pixels/
// Author: Gary McCann

const robot = await npm ("robotjs");

const mousePosition = robot.getMousePos();
const screen = robot.getScreenSize();

robot.moveMouse(screen.width - 354, 47);

setTimeout(() => {
  robot.mouseClick();
  robot.moveMouse(mousePosition.x, mousePosition.y);
}, 100);
